src/Foundatio/Storage/FolderFileStorage.cs

```diff
@@ -77,22 +81,32 @@ public class FolderFileStorage : IFileStorage {
             path = path.NormalizePath();
 
             string file = Path.Combine(Folder, path);
-            string directory = Path.GetDirectoryName(file);
-            if (directory != null)
-                Directory.CreateDirectory(directory);
 
             try {
-                using (var fileStream = File.Create(file)) {
+                using (var fileStream = CreateFileStream(file)) {
                     if (stream.CanSeek)
                         stream.Seek(0, SeekOrigin.Begin);
 
                     stream.CopyTo(fileStream);
                 }
-            } catch (Exception) {
+            } catch (Exception ex) {
+                _logger.LogError(ex, "Error trying to save file '{path}'.", path);
                 return Task.FromResult(false);
             }
 
             return Task.FromResult(true);
+
+            Stream CreateFileStream(string filePath) {
+                try {
+                    return File.Create(filePath);
+                }
+                catch (DirectoryNotFoundException) {
```

Comment:
What happens if the file already exists? I'm thinking this code could be outside of the catch,

Reply:
I have checked for the APIs of file operations, the only is to use File.Exist check, but it leads more I/O operations.

Reply:
It will override the existing files.